pilot/pkg/networking/plugin/authz/authorization.go

```diff
@@ -0,0 +1,137 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package authz converts Istio RBAC (role-based-access-control) policies (ServiceRole and ServiceRoleBinding)
+// to the Envoy RBAC filter config to enforce access control to the service co-located with Envoy.
+// The generation is controlled by ClusterRbacConfig (a singleton custom resource with cluster scope).
+// User could disable this plugin by either deleting the ClusterRbacConfig or set the ClusterRbacConfig.mode
+// to OFF.
+// Note: ClusterRbacConfig is not created with istio installation which means this plugin doesn't
+// generate any RBAC config by default.
+package authz
+
+import (
+	xdsapi "github.com/envoyproxy/go-control-plane/envoy/api/v2"
+
+	"istio.io/istio/pilot/pkg/model"
+	"istio.io/istio/pilot/pkg/networking/plugin"
+	"istio.io/istio/pilot/pkg/networking/util"
+	authz_builder "istio.io/istio/pilot/pkg/security/authz/builder"
+	istiolog "istio.io/pkg/log"
+)
+
+var (
+	rbacLog = istiolog.RegisterScope("rbac", "rbac debugging", 0)
+)
+
+// Plugin implements Istio Authorization
+type Plugin struct{}
+
+// NewPlugin returns an instance of the authorization plugin
+func NewPlugin() plugin.Plugin {
+	return Plugin{}
+}
+
+// OnOutboundListener is called whenever a new outbound listener is added to the LDS output for a given service
+// Can be used to add additional filters on the outbound path
+func (Plugin) OnOutboundListener(in *plugin.InputParams, mutable *plugin.MutableObjects) error {
+	if in.Node.Type != model.Router {
+		// Only care about router.
+		return nil
+	}
+
+	return buildFilter(in, mutable)
+}
+
+// OnInboundFilterChains is called whenever a plugin needs to setup the filter chains, including relevant filter chain configuration.
+func (Plugin) OnInboundFilterChains(in *plugin.InputParams) []plugin.FilterChain {
+	return nil
+}
+
+// OnInboundListener is called whenever a new listener is added to the LDS output for a given service
+// Can be used to add additional filters (e.g., mixer filter) or add more stuff to the HTTP connection manager
+// on the inbound path
+func (Plugin) OnInboundListener(in *plugin.InputParams, mutable *plugin.MutableObjects) error {
+	if in.Node.Type != model.SidecarProxy {
+		// Only care about sidecar.
+		return nil
+	}
+
+	return buildFilter(in, mutable)
+}
+
+func buildFilter(in *plugin.InputParams, mutable *plugin.MutableObjects) error {
```

Comment:
seems this func never returns err, shall we just remove or propagate err say from ```NewBuilder``` ?

Reply:
Currently we don't block the filter building in pilot, so I just removed the `error` part to keep the same behavior.